common/lib/capa/capa/responsetypes.py

```diff
@@ -720,14 +726,103 @@ def assign_choice_names(self):
 
     def get_score(self, student_answers):
 
+        """
+        Answer lists:
+          all_choices: the full list of checkboxes
+          student_answer: what the student actually chose (note no "s")
+          student_non_answer: what they didn't choose
+          self.correct_choices: boxes that should be checked
+          self.incorrect_choices: boxes that should NOT be checked
+        """
+        
+        # pipe for union of sets
+        all_choices = self.correct_choices | self.incorrect_choices   
+        
         student_answer = student_answers.get(self.answer_id, [])
 
         if not isinstance(student_answer, list):
             student_answer = [student_answer]
 
         no_empty_answer = student_answer != []
+        # "None apply" should really be a valid choice for "check all that apply",
+        # but it throws an error if all the checks are blank.
```

Comment:
I just don't get how this comment, in this location, relates to the code
around it.

On Tue, Jun 23, 2015 at 2:16 PM, Colin-Fredericks <notifications@github.com>
wrote:

> In common/lib/capa/capa/responsetypes.py
> <https://github.com/edx/edx-platform/pull/8574#discussion_r33073112>:
>
> >          student_answer = student_answers.get(self.answer_id, [])
> >
> >          if not isinstance(student_answer, list):
> >              student_answer = [student_answer]
> >
> >          no_empty_answer = student_answer != []
> > +        # "None apply" should really be a valid choice for "check all that apply",
> > +        # but it throws an error if all the checks are blank.
>
> No, I added that because I initially went in and said "Why is this stupid
> restriction here?" And I still don't know why, really, but it keeps things
> from breaking, and I wanted to warn other folks who might have the same
> thought. I can remove the comment if you like.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/8574/files#r33073112>.
>